### PR TITLE
Add `X-Request-ID` to requests before we have a transaction ID

### PR DESCRIPTION
**Pull request by mcmcgrath13**
Make sure we can trace the request to infra.

RAICode test branch: https://github.com/RelationalAI/raicode/pull/13681

### CODE REVIEW

**Review by larf311 — APPROVED** [review]

**Comment by mcmcgrath13** [comment]
This is blocked until https://relationalai.atlassian.net/browse/RAI-11293 is complete

File: src/testrel.jl [hunk 1]
```diff
@@ -420,7 +420,12 @@ function _test_rel_steps(;
             @testset TestRelTestSet nested=nested "$name" begin
                 create_test_database(schema, clone_db)
                 for (index, step) in enumerate(steps)
-                    _test_rel_step(index, step, schema, test_engine, name, length(steps))
+                    inner_ts = _test_rel_step(index, step, schema, test_engine, name, length(steps))
+                    # short circuit if something errored
+                    if anyerror(inner_ts) && index < length(steps)
+                        @error "Error running $name - not executing further steps"
```
Comment (mmcgr) [inline src/testrel.jl:426]: I'm not against changing test behaviour to stopping if errors are encountered, but the error message could be stronger, something like "Terminal error running...", to make it clear that further testing is pointless.

File: src/testrel.jl [hunk 2]
```diff
@@ -538,12 +548,15 @@ function _execute_test(
...
 )
-    @debug("$name: Starting execution")
+    request_id = string(UUIDs.uuid4())
+    @debug("$name: Starting execution ReqId=$request_id")
     rsp = try
         # Exec async really should return after 2-3 seconds
-        exec_async(context, schema, engine, program; readtimeout=30, readonly)
+        headers = ["X-Request-ID" => request_id]
+        exec_async(context, schema, engine, program; readtimeout=30, readonly, headers)
+        throw("uh oh")
```
Comment (mcmcgrath13) [inline src/testrel.jl:557]:
```suggestion
```

File: src/testrel.jl [hunk 3]
```diff
@@ -431,7 +431,12 @@ function _test_rel_steps(;
             @testset TestRelTestSet nested = nested "$name" begin
                 create_test_database(schema, clone_db)
                 for (index, step) in enumerate(steps)
-                    _test_rel_step(index, step, schema, test_engine, name, length(steps))
+                    inner_ts = _test_rel_step(index, step, schema, test_engine, name, length(steps))
+                    # short circuit if something errored
+                    if anyerror(inner_ts) && index < length(steps)
+                        @error "Error running $name - not executing further steps"
```
Comment (mcmcgrath13) [inline src/testrel.jl:437]:
```suggestion
                        @error "Fatal error running $name - not executing further steps"
```

File: src/testrel.jl [hunk 4]
```diff
@@ -554,6 +554,7 @@ function _execute_test(
         # Exec async really should return after 2-3 seconds
         headers = ["X-Request-ID" => request_id]
         exec_async(context, schema, engine, program; readtimeout=30, readonly, headers)
+        throw("uh oh")
```
Comment (larf311) [inline src/testrel.jl]: Was this done so the automated tests run? Or maybe to get a deployable image? I'm curious b/c we'll soon (hopefully) start looking at improving the developer inner loop.

**Review by mmcgr — APPROVED** [review]

**Pull request merged**